lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -0,0 +1,538 @@
+                self.charm.status.set(WaitingStatus("s3-broken event: cluster not ready yet"))
+            else:
+                self.charm.status.set(
+                    BlockedStatus("Unexpected error during plugin configuration, check the logs")
+                )
+                # There was an unexpected error, log it and block the unit
+                logger.error(e)
+            self.charm.plugin_manager.reset_event_scope()
+            event.defer()
+            return
+        # we need to do this task, so we will ask for an exception from lint
+        except:  # noqa: E722
+            self.charm.plugin_manager.reset_event_scope()
+            return
+        self.charm.plugin_manager.reset_event_scope()
+        self.charm.status.clear(ActiveStatus())
+
+    def _execute_s3_broken_calls(self):
+        """Executes the s3 broken API calls."""
+        return  # do not execute anything as we intend to keep the backups untouched
+
+    def _check_repo_status(self) -> BackupServiceState:
+        try:
+            return self.get_service_status(self._request("GET", f"_snapshot/{S3_REPOSITORY}"))
+        except (ValueError, OpenSearchHttpError, requests.HTTPError):
+            return BackupServiceState.RESPONSE_FAILED_NETWORK
+
+    def _check_snapshot_status(self) -> BackupServiceState:
+        try:
+            return self.get_snapshot_status(self._request("GET", "/_snapshot/_status"))
+        except (ValueError, OpenSearchHttpError, requests.HTTPError):
+            return BackupServiceState.RESPONSE_FAILED_NETWORK
+
+    def _register_snapshot_repo(self) -> BackupServiceState:
+        """Registers the snapshot repo in the cluster."""
+        info = self.s3_client.get_s3_connection_info()
+        return self.get_service_status(
+            self._request(
+                "PUT",
+                f"_snapshot/{S3_REPOSITORY}",
+                payload={
+                    "type": "s3",
+                    "settings": {
+                        "region": info.get("region"),
+                        "endpoint": info.get("endpoint"),
+                        "protocol": "http",  # TODO: roll back to https once we have the cert
+                        "bucket": info["bucket"],
+                        "base_path": info.get("path", S3_REPO_BASE_PATH),
+                    },
+                },
+            )
+        )
+
+    def _is_s3_fully_configured(self) -> bool:
+        """Checks if relation is set and all configs needed are present.
+
+        The get_s3_connection_info() checks if the relation is present, and if yes,
+        returns the data in it.
+
+        This method will go over the output and generate a list of missing parameters
+        that are manadatory to have be present. An empty list means everything is present.
+        """
+        missing_s3_configs = [
+            config
+            for config in ["region", "bucket", "access-key", "secret-key"]
```

Comment:
region is not obligatory in some cases.
see examples https://charmhub.io/postgresql-k8s/docs/h-configure-s3-radosgw
can be separate PR